File: test_cloud_api.py
#!/usr/bin/env python3
import requests
import json

# 服务器配置
DOMAIN = "http://emagen.323424.xyz"
API_BASE = f"{DOMAIN}/api/v1"

# 测试计数
total_tests = 0
passed_tests = 0

def test_api(name, method, url, data=None, headers=None):
    """测试单个API"""
    global total_tests, passed_tests
    total_tests += 1
    
    try:
        if method == "GET":
            response = requests.get(url, headers=headers)
        elif method == "POST":
            response = requests.post(url, json=data, headers=headers)
        elif method == "PUT":
            response = requests.put(url, json=data, headers=headers)
        elif method == "DELETE":
            response = requests.delete(url, headers=headers)
        
        if response.status_code in [200, 201]:
            passed_tests += 1
            print(f"✅ {name}: 成功 (状态码: {response.status_code})")
            return response.json()
        else:
            print(f"❌ {name}: 失败 (状态码: {response.status_code})")
            print(f"   响应: {response.text}")
            return None
    except Exception as e:
        print(f"❌ {name}: 错误 - {str(e)}")
        return None

def test_frontend():
    """测试前端页面"""
    print("\n📄 测试前端页面...")
    
    pages = [
        ("首页", f"{DOMAIN}/"),
        ("管理后台", f"{DOMAIN}/admin.html")
    ]
    
    for name, url in pages:
        try:
            response = requests.get(url)
            if response.status_code == 200:
                print(f"✅ {name}: 可访问")
            else:
                print(f"❌ {name}: 状态码 {response.status_code}")
        except Exception as e:
            print(f"❌ {name}: 错误 - {str(e)}")

def main():
    print("🚀 开始测试云服务器API...")
    print(f"🌐 服务器地址: {DOMAIN}")
    
    # 1. 测试客户端API
    print("\n=== 客户端API测试 ===")
    
    # 获取门店列表
    stores = test_api("获取门店列表", "GET", f"{API_BASE}/client/stores")
    
    # 获取技师列表
    therapists = test_api("获取技师列表", "GET", f"{API_BASE}/client/therapists/search")
    
    # 2. 测试管理端API
    print("\n=== 管理端API测试 ===")
    
    # 管理员登录
    login_data = {"username": "admin", "password": "admin123"}
    login_result = test_api("管理员登录", "POST", f"{API_BASE}/admin/login", login_data)
    
    if login_result and login_result.get("success"):
        token = login_result["data"]["token"]
        headers = {"Authorization": f"Bearer {token}"}
        
        # 创建测试门店
        store_data = {
            "name": "测试门店",
            "address": "测试地址123号",
            "phone": "13800138000",
            "business_hours": "9:00-21:00",
            "description": "这是一个测试门店"
        }
        store_result = test_api("创建门店", "POST", f"{API_BASE}/admin/stores", store_data, headers)
        
        if store_result and store_result.get("success"):
            store_id = store_result["data"]["id"]
            
            # 创建测试技师
            therapist_data = {
                "store_id": store_id,
                "name": "测试技师",
                "position": "推拿师",
                "years_of_experience": 5,
                "specialties": ["按摩", "推拿"],
                "phone": "13900139000"
            }
            therapist_result = test_api("创建技师", "POST", f"{API_BASE}/admin/therapists", therapist_data, headers)
        
        # 获取统计数据
        test_api("获取统计概览", "GET", f"{API_BASE}/admin/statistics/overview", headers=headers)
    
    # 3. 测试前端页面
    test_frontend()
    
    # 显示测试结果
    print(f"\n=== 测试结果 ===")
    print(f"总测试数: {total_tests}")
    print(f"通过: {passed_tests}")
    print(f"失败: {total_tests - passed_tests}")
    print(f"成功率: {(passed_tests/total_tests*100):.2f}%")

if __name__ == "__main__":
    main()

File: test-cloud-api.js
const http = require('http');
const querystring = require('querystring');

// 云端API配置
const API_HOST = 'emagen.323424.xyz';
const API_PORT = 80;
const API_BASE = '/api/v1';

// 测试结果统计
const testResults = {
    total: 0,
    passed: 0,
    failed: 0,
    details: []
};

// HTTP请求封装
function makeRequest(options, data = null) {
    return new Promise((resolve, reject) => {
        const req = http.request({
            hostname: API_HOST,
            port: API_PORT,
            ...options
        }, (res) => {
            let responseData = '';
            res.on('data', (chunk) => responseData += chunk);
            res.on('end', () => {
                try {
                    const result = {
                        status: res.statusCode,
                        headers: res.headers,
                        data: responseData ? JSON.parse(responseData) : null
                    };
                    resolve(result);
                } catch (error) {
                    resolve({
                        status: res.statusCode,
                        headers: res.headers,
                        data: responseData
                    });
                }
            });
        });

        req.on('error', reject);
        
        if (data) {
            req.write(JSON.stringify(data));
        }
        
        req.end();
    });
}

// 记录测试结果
function recordTest(name, endpoint, result, details) {
    testResults.total++;
    if (result) {
        testResults.passed++;
        console.log(`✅ ${name}`);
        console.log(`   ${endpoint}`);
        console.log(`   ${details}`);
    } else {
        testResults.failed++;
        console.log(`❌ ${name}`);
        console.log(`   ${endpoint}`);
        console.log(`   ${details}`);
    }
    console.log('');
}

// 运行云端API测试
async function testCloudAPIs() {
    console.log('🌐 测试云端API...');
    console.log(`服务器: http://${API_HOST}`);
    console.log(`时间: ${new Date().toLocaleString('zh-CN')}\n`);

    // 1. 健康检查
    try {
        const res = await makeRequest({
            method: 'GET',
            path: '/health'
        });
        recordTest(
            '健康检查',
            'GET /health',
            res.status === 200,
            res.status === 200 ? `服务正常 - ${res.data.service}` : `状态码: ${res.status}`
        );
    } catch (error) {
        recordTest('健康检查', 'GET /health', false, error.message);
    }

    // 2. 获取门店列表
    try {
        const res = await makeRequest({
            method: 'GET',
            path: `${API_BASE}/client/stores`
        });
        recordTest(
            '获取门店列表',
            'GET /api/v1/client/stores',
            res.status === 200 && res.data.success,
            res.data.success ? `成功 - ${res.data.data.stores.length} 家门店` : `失败 - 状态码: ${res.status}`
        );
    } catch (error) {
        recordTest('获取门店列表', 'GET /api/v1/client/stores', false, error.message);
    }

    // 3. 搜索技师
    try {
        const res = await makeRequest({
            method: 'GET',
            path: `${API_BASE}/client/therapists/search?limit=5`
        });
        recordTest(
            '搜索技师',
            'GET /api/v1/client/therapists/search',
            res.status === 200 && res.data.success,
            res.data.success ? `成功 - ${res.data.data.therapists.length} 位技师` : `失败 - 状态码: ${res.status}`
        );
    } catch (error) {
        recordTest('搜索技师', 'GET /api/v1/client/therapists/search', false, error.message);
    }

    // 4. 技师搜索（按专长）
    try {
        const params = querystring.stringify({ specialty: '艾灸' });
        const res = await makeRequest({
            method: 'GET',
            path: `${API_BASE}/client/therapists/search?${params}`
        });
        recordTest(
            '搜索技师（按专长）',
            'GET /api/v1/client/therapists/search?specialty=艾灸',
            res.status === 200 && res.data.success,
            res.data.success ? `成功 - 找到 ${res.data.data.therapists.length} 位擅长艾灸的技师` : `失败 - 状态码: ${res.status}`
        );
    } catch (error) {
        recordTest('搜索技师（按专长）', 'GET /api/v1/client/therapists/search?specialty=艾灸', false, error.message);
    }

    // 5. 管理员登录测试
    try {
        const res = await makeRequest({
            method: 'POST',
            path: `${API_BASE}/admin/login`,
            headers: { 'Content-Type': 'application/json' }
        }, {
            username: 'admin',
            password: 'admin123'
        });
        recordTest(
            '管理员登录',
            'POST /api/v1/admin/login',
            res.status === 200 && res.data.success,
            res.data.success ? '成功 - 获得认证令牌' : `失败 - 状态码: ${res.status}`
        );
    } catch (error) {
        recordTest('管理员登录', 'POST /api/v1/admin/login', false, error.message);
    }

    // 6. 测试前端页面
    try {
        const res = await makeRequest({
            method: 'GET',
            path: '/frontend/index.html'
        });
        recordTest(
            '客户端页面',
            'GET /frontend/index.html',
            res.status === 200,
            res.status === 200 ? '页面可访问' : `状态码: ${res.status}`
        );
    } catch (error) {
        recordTest('客户端页面', 'GET /frontend/index.html', false, error.message);
    }

    // 7. 测试管理端页面
    try {
        const res = await makeRequest({
            method: 'GET',
            path: '/frontend/admin.html'
        });
        recordTest(
            '管理端页面',
            'GET /frontend/admin.html',
            res.status === 200,
            res.status === 200 ? '页面可访问' : `状态码: ${res.status}`
        );
    } catch (error) {
        recordTest('管理端页面', 'GET /frontend/admin.html', false, error.message);
    }

    // 统计结果
    console.log('========== 测试统计 ==========');
    console.log(`总测试数: ${testResults.total}`);
    console.log(`✅ 通过: ${testResults.passed}`);
    console.log(`❌ 失败: ${testResults.failed}`);
    console.log(`成功率: ${((testResults.passed / testResults.total) * 100).toFixed(2)}%`);
}

// 运行测试
testCloudAPIs().catch(console.error);